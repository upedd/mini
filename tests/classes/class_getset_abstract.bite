native print;

class Foo {
    abstract get x;

    abstract set z;

    abstract y;

    display() {
        y = 40;
        z = 30;
        print(x);
        print(y);
    }
}

class Bar : Foo {
    override get x {
        print("bar.x");
        return 10;
    }

    override get y {
        print("bar.y");
        return 20;
    }

    override set y(value) {
        print("=bar.y");
    }

    get z {
        print("bar.z");
        return 30;
    }

    override set z(value) {
        print("bar.z");
    }
}

let instance = Bar();
print(instance.x);
instance.y = 10;
print(instance.y);
instance.z = 20;
print(instance.z);
instance.display();
