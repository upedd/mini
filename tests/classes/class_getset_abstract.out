bar.x
10
=bar.y
bar.y
20
=bar.z
bar.z
30
=bar.y
=bar.z
bar.x
10
bar.y
20